### PR TITLE
test: timezone with data_histogram and composite date_histogram

### PR DESCRIPTION
**Pull request by salvatore-campagna**
This test checks that time zones are handled correctly by `data histogram` 
and `composite date histogram`. If the query includes a specific time zone
dates and times should be returned with the right time difference according
to the time zone requested by the user.

Issue: #49357

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/490_date_histogram_timezone.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                  - date_field:
+                      date_histogram:
+                        field: date
+                        format: "yyyy-MM-dd HH:mm:ss"
+                        calendar_interval: hour
+
+  - match: { hits.total.value: 16 }
+  - match: { hits.total.relation: "eq" }
+  - length: { aggregations.date_histogram_with_tz.buckets: 4 }
+  - match: { aggregations.date_histogram_with_tz.buckets.0.key_as_string: "2021-05-02 03:00:00" }
```
Comment (nik9000) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/490_date_histogram_timezone.yml:96]: I wonder if you can make the names the same width and check the two bucket starts side by side so it's really easy to see the time zone effects. Like:
```
   - match: { aggregations.yes_tz.buckets.0.key_as_string: "2021-05-02 03:00:00" }
   - match: { aggregations.no__tz.buckets.0.key_as_string: "2021-05-02 04:00:00" }
```

**Review by nik9000 — COMMENTED** [review]
Could you stick this in `360_date_histogram.yml`?

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml [hunk 2]
```diff
@@ -61,3 +112,103 @@ setup:
...
+                field: date
+                format: "yyyy-MM-dd HH:mm:ss"
+                calendar_interval: hour
+                time_zone: Asia/Jakarta
+            date_histogram_simple_no__tz:
+              date_histogram:
+                field: date
+                format: "yyyy-MM-dd HH:mm:ss"
+                calendar_interval: hour
+            date_histogram_compos_yes_tz:
```
Comment (not-napoleon) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml:138]: The composite version should be in the composite test file, `230_composite.yml`, please.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml [hunk 3]
```diff
@@ -61,3 +112,103 @@ setup:
   - match: { aggregations.histo.buckets.8.doc_count: 1 }
   - match: { aggregations.histo.buckets.12.key_as_string: "2016-06-01T00:00:00.000Z" }
   - match: { aggregations.histo.buckets.12.doc_count: 1 }
+
+---
+"date_histogram and date_histogram_composite timezone":
+  - skip:
+      version: " - 8.1.99"
+      reason: bug fixed in 8.2.0
```
Comment (not-napoleon) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/360_date_histogram.yml:120]: Is this really fixed in 8.2?  I can understand not wanting to bisect all the way back to 7.2 to see where it got fixed, but if it works in 7.17, let's at least enable this for the LTS release, please.  That way it'll run with BWC stuff.  A test-only backport is probably a good idea too.

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Review by not-napoleon — APPROVED** [review]
LGTM

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Comment by salvatore-campagna** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.1|[](https://github.com/elastic/elasticsearch/pull/85466)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/85467)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**